clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.loadbalancer
+
+import com.netflix.spinnaker.clouddriver.openstack.client.OpenstackClientProvider
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.loadbalancer.OpenstackLoadBalancerDescription
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackOperationException
+import com.netflix.spinnaker.clouddriver.openstack.domain.LoadBalancerPool
+import com.netflix.spinnaker.clouddriver.openstack.domain.Region
+import com.netflix.spinnaker.clouddriver.openstack.domain.VirtualIP
+import com.netflix.spinnaker.clouddriver.openstack.task.TaskAware
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
+import org.openstack4j.model.network.NetFloatingIP
+import org.openstack4j.model.network.ext.LbPool
+import org.openstack4j.model.network.ext.Vip
+
+class UpsertOpenstackLoadBalancerAtomicOperation implements AtomicOperation<Map>, TaskAware {
```

Comment:
TaskAware is a trait that encapsulates task status behavior so that it can be used across multiple concrete classes ... where ever TaskUpdates need to happen.